packages/api/src/platforms/vtex/utils/shouldUpdateShippingData.ts

```diff
@@ -0,0 +1,156 @@
+    return false
+  }
+
+  if (!hasSessionPostalCodeOrGeoCoordinates(session)) {
+    return false
+  }
+
+  const { address, logisticsInfo } = orderForm.shippingData!
+
+  if (checkPostalCode(address, session.postalCode)) {
+    return true
+  }
+
+  if (checkGeoCoordinates(address, session.geoCoordinates)) {
+    return true
+  }
+
+  if (shouldUpdateDeliveryChannel(logisticsInfo, session)) {
+    return true
+  }
+
+  if (shouldUpdateDeliveryMethod(logisticsInfo, session)) {
+    return true
+  }
+
+  if (shouldUpdateDeliveryWindow(logisticsInfo, session)) {
+    return true
+  }
+  return false
+}
+
+// Validate if theres any item inside the orderForm
+const hasItems = (orderForm: OrderForm) => {
+  return orderForm.items.length !== 0
+}
+
+// Validate if theres any postal Code or GeoCoordinates set at the session
+const hasSessionPostalCodeOrGeoCoordinates = (session: IStoreSession) => {
+  return (
+    !!session.postalCode ||
+    (session.geoCoordinates &&
+      session.geoCoordinates.latitude &&
+      session.geoCoordinates.longitude)
+  )
+}
+
+// Validate if theres a difference between the session postal code and orderForm postal code
+const checkPostalCode = (
+  address: CheckoutAddress | null | undefined,
+  postalCode: string | null | undefined
+) => {
+  return typeof postalCode === 'string' && address?.postalCode !== postalCode
+}
+
+// Validate if theres a difference between the session geoCoords and orderForm geoCoords
+const checkGeoCoordinates = (
+  address: CheckoutAddress | null | undefined,
+  geoCoordinates: IStoreGeoCoordinates | null | undefined
+) => {
+  return (
+    typeof geoCoordinates?.latitude === 'number' &&
+    typeof geoCoordinates?.longitude === 'number' &&
+    (address?.geoCoordinates[0] !== geoCoordinates?.longitude ||
+      address?.geoCoordinates[1] !== geoCoordinates?.latitude)
+  )
+}
+
+// Validate if the deliveryChannel from the session is different from the selected delivery channel
+// and if so needs to validate if the deliveryChannel for the session is available inside the slas for the item
+const shouldUpdateDeliveryChannel = (
+  logisticsInfo: LogisticsInfo[],
+  session: IStoreSession | null | undefined
+) => {
+  if (!session?.deliveryMode?.deliveryChannel) {
+    return false
+  }
+  const { deliveryChannel } = session.deliveryMode
+  for (const item of logisticsInfo) {
+    if (item.selectedDeliveryChannel !== deliveryChannel) {
+      const matchingSla = item.slas.find(
+        (sla) => sla.deliveryChannel === deliveryChannel
+      )
+      if (matchingSla) {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+// Validate if the deliveryMethod from the session is different from the selectedSLA
+// and if so needs to validate if the deliveryMethod for the session is available inside the slas for the item
+const shouldUpdateDeliveryMethod = (
+  logisticsInfo: LogisticsInfo[],
+  session: IStoreSession
+) => {
+  if (!session?.deliveryMode?.deliveryMethod) {
```

Comment:
is `deliveryMethod` a string or a number? Asking this because if it is a number, there's the possibility of it being `0` and being unintentionally discarded by this condition.

Reply:
deliveryMethod is always a string!

Reference: https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/attachments/shippingData
selectedSla object